### PR TITLE
imgproxy: init at 2.8.1

### PR DESCRIPTION
**Pull request by paluh**
###### Motivation for this change

[imgproxy](https://imgproxy.net/) server is not available in nixpkgs. I've packaged only module. I'm not sure if I should also try to provide appropriate service package (this is my first package).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by erikarvstedt** [comment]
Please add these fixes: https://github.com/erikarvstedt/nixpkgs/commit/4648cef5b92662669754e3c0da8965f7a5005a2c

**Comment by paluh** [comment]
> Welcome to nixpkgs! ￼

Thanks! :-)

I've merged your fixes directly. Do we want to have this merge commit in history or should I squash it into single one and push?

I've rerun `nix-shell -p nix-review --run "nix-review rev HEAD"` and everything seems fine.

**Comment by erikarvstedt** [comment]
Please squash it.

**Comment by paluh** [comment]
> Please squash it.

No problem. Done :-)
I've rerun the tests and pushed squased version.

P.S. I'm not sure why github claims that I've requested some reviews and incorporates some additional commit in this PR...

**Comment by paluh** [comment]
One sec. I've probably listed this additional commit during my rebase. Going to fix it...

**Comment by paluh** [comment]
Cleaned up. Sorry for this additional spam :-)

**Comment by erikarvstedt** [comment]
This looks great now, congrats! We'll now have to wait for a committer to merge this.

File: pkgs/servers/imgproxy/default.nix
Comment (erikarvstedt) [inline pkgs/servers/imgproxy/default.nix]: Woops, forgot this one: Please move `pname` above `version`.

File: pkgs/servers/imgproxy/default.nix
Comment (erikarvstedt) [inline pkgs/servers/imgproxy/default.nix]: Add a space before `"`

**Comment by paluh** [comment]
Fixed and tested. @erikarvstedt thanks a lot for your time, review and support!

File: pkgs/servers/imgproxy/default.nix
Comment (erikarvstedt) [inline pkgs/servers/imgproxy/default.nix]: `modSha256` is not needed

File: pkgs/servers/imgproxy/default.nix
Comment (erikarvstedt) [inline pkgs/servers/imgproxy/default.nix]: `goPackagePath` can be removed.

File: pkgs/servers/imgproxy/default.nix
Comment (erikarvstedt) [inline pkgs/servers/imgproxy/default.nix]: Please put `modSha256` below the `src` definition.

**Comment by erikarvstedt** [comment]
Sorry, my go-package-fu is a bit rusty, otherwise I would have included all these changes in my inital patch.

File: pkgs/servers/imgproxy/default.nix
Comment (paluh) [reply pkgs/servers/imgproxy/default.nix]: I've just dropped them and running test build on my machine. Should I bring them back?

Comment (erikarvstedt) [reply pkgs/servers/imgproxy/default.nix]: Sorry, my deleted comment was of course an error, I meant that `goPackagePath` can be removed.

File: pkgs/servers/imgproxy/default.nix
Comment (erikarvstedt) [inline pkgs/servers/imgproxy/default.nix]: Add one newline before `modSha256`

File: pkgs/servers/imgproxy/default.nix
Comment (erikarvstedt) [inline pkgs/servers/imgproxy/default.nix]:
```
owner = pname;
repo = pname;
```

**Comment by paluh** [comment]
> Sorry, my go-package-fu is a bit rusty (...)

No problem. I've removed `goPackagePath` but `modSha256` seems to be required (test build failes with `(..)called without required argument 'modSha256'). I've moved it as you suggested just below `src`.

**Comment by erikarvstedt** [comment]
Awesome, it's perfect now. Thanks for you patience! :smiley:

**Comment by paluh** [comment]
I've rerun test build. Everything is working fine.

> Awesome, it's perfect now. Thanks for you patience! 

I really appreciate your detailed review. Thanks again!

**Comment by erikarvstedt** [comment]
@grahamcofborg build imgproxy

**Review by tomberek — APPROVED** [review]
Builds on NixOS. Functions as expected.

**Review by stigtsp — APPROVED** [review]
Looks good to me

**Review by erikarvstedt — APPROVED** [review]

**Comment by paluh** [comment]
@erikarvstedt @stigtsp
Do you think that providing `service` expression for this app would be valuable? If so should I include it in this commit or can I do it in a separate one?

**Comment by erikarvstedt** [comment]
It's best to do it in a separate PR.

**Comment by paluh** [comment]
> It's best to do it in a separate PR.

Cool. I'm going to work on it along with the coding which I do for my internal project deployment.

File: maintainers/maintainer-list.nix
Comment (kalbasit) [inline maintainers/maintainer-list.nix]: Please put this in a different commit.

File: pkgs/servers/imgproxy/default.nix
Comment (kalbasit) [inline pkgs/servers/imgproxy/default.nix]:
```suggestion
    homepage = "https://imgproxy.net";
```

**Review by doronbehar — APPROVED** [review]

File: pkgs/servers/imgproxy/default.nix
Comment (Infinisil) [reply pkgs/servers/imgproxy/default.nix]: Fwiw, I've stopped complaining about tiny issues like this, since I've been told there's too much bikeshedding like that, which can be demotivating for contributors.

**Pull request merged**
